src/test/java/org/opensearch/security/dlic/rest/api/UserApiTest.java

```diff
@@ -546,6 +548,104 @@ public void testPasswordRules() throws Exception {
+
+        Settings nodeSettings =
+                Settings.builder()
+                        .put(ConfigConstants.SECURITY_RESTAPI_PASSWORD_VALIDATION_ERROR_MESSAGE, "xxx")
+                        .put(ConfigConstants.SECURITY_RESTAPI_PASSWORD_SCORE_BASED_VALIDATION, "true")
+                        .put(
+                                ConfigConstants.SECURITY_RESTAPI_PASSWORD_SCORE_BASED_VALIDATION_MIN_PASSWORD_LENGTH,
+                                "2")
+                        .put(
+                                ConfigConstants.SECURITY_RESTAPI_PASSWORD_SCORE_BASED_VALIDATION_SCORE_STRENGTH,
+                                PasswordValidationScoreStrength.STRONG.name().toLowerCase(Locale.ROOT)
+                        ).build();
+
+        setup(nodeSettings);
+
+        rh.keystore = "restapi/kirk-keystore.jks";
+        rh.sendAdminCertificate = true;
+
+        // initial configuration, 6 users
+        HttpResponse response = rh
+                .executeGetRequest("_plugins/_security/api/" + CType.INTERNALUSERS.toLCString());
+        Assert.assertEquals(response.getBody(), HttpStatus.SC_OK, response.getStatusCode());
+        Settings settings = Settings.builder().loadFromSource(response.getBody(), XContentType.JSON).build();
+        Assert.assertEquals(133, settings.size());
+
+        addUserWithPassword("tooshoort", "", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("tooshoort", "123", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("tooshoort", "1234567", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("tooshoort", "1Aa%", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("no-nonnumeric", "123456789", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("no-uppercase", "a123456789", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("no-lowercase", "A123456789", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("ok1", "a%A123456789", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("ok2", "$aA123456789", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("ok3", "$Aa123456789", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("ok4", "$1aAAAAAAAAA", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("empty_password_no_hash", "", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPasswordAndHash("empty_password", "", "$%^123", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPasswordAndHash("null_password", null, "$%^123", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("asd@asd.com", "ASD@asdCOMsimilar", HttpStatus.SC_BAD_REQUEST);
+        addUserWithPassword("asd@asd.com", "asdasd3dasdadsdsadasdssd@DSADSD", HttpStatus.SC_CREATED);
+
+        response = rh.executePatchRequest(PLUGINS_PREFIX + "/api/internalusers", "[{ \"op\": \"add\", \"path\": \"/ok4\", \"value\": {\"password\": \"bla\", \"backend_roles\": [\"vulcan\"] } }]", new Header[0]);
+        Assert.assertEquals(HttpStatus.SC_BAD_REQUEST, response.getStatusCode());
```

Comment:
Could you add a helper function for these scenarios, seems like its a repeat of the `addUserWithPassword` cases above, but harder to read as is.